publish_to_hugging_face.py

```diff
@@ -0,0 +1,65 @@
+import argparse
+import logging
+import sys
+
+from app_utils import hugging_face_utils
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="")
+    parser.add_argument(
+        "-e",
+        "--experiment_name",
+        required=True,
+        help="experiment name",
+        default=argparse.SUPPRESS,
+    )
+
+    parser.add_argument(
+        "-d",
+        "--device",
+        help="'cpu' or 'cuda:0', if the GPU device id is 0",
+        default="cuda:0",
+    )
+
+    parser.add_argument(
+        "-a",
+        "--api_key",
+        required=False,
+        help="Hugging Face API Key",
+        default=argparse.SUPPRESS,
+    )
+
+    parser.add_argument(
+        "-u",
+        "--user_id",
+        required=False,
+        help="Hugging Face User ID",
+        default=argparse.SUPPRESS,
+    )
+
+    parser.add_argument(
+        "-m",
+        "--model_name",
+        required=True,
+        help="Hugging Face Model Name",
+        default=argparse.SUPPRESS,
+    )
+
+    parser_args, unknown = parser.parse_known_args(sys.argv)
+
+    experiment_name = parser_args.experiment_name
+    device = parser_args.device
+    api_key = parser_args.api_key
+    user_id = parser_args.user_id
```

Comment:
this fails, if no api_key is provided (as it is optional)

please also check for the rest

Reply:
checking for the optional variables of api_key, user_id, and model_name.
   api_key = parser_args.api_key if parser_args.api_key is not None else ""
    user_id = parser_args.user_id if parser_args.user_id is not None else ""
    model_name = parser_args.model_name if parser_args.model_name is not None else ""